frontend/store/surplus.ts

```diff
@@ -112,7 +131,19 @@ export const actions = {
         try {
             commit('setAuctionsFetching', true);
             const auctions = await fetchActiveSurplusAuctions(network);
-            auctions.forEach(auction => commit('addAuctionToStorage', auction));
+            const collectedAuctions: SurplusAuctionCollected[] = [];
+            const activeAuctions: SurplusAuctionActive[] = [];
+            auctions.forEach(auc => {
+                if (auc.state === "collected") {
+                    collectedAuctions.push(auc)
+                } else {
+                    activeAuctions.push(auc)
+                }
+            })
+            const enrichedAuctionPromises = activeAuctions.map(auc => enrichSurplusAuction(network, auc));
+            const auctionTransactions = await Promise.all(enrichedAuctionPromises);
+            auctionTransactions.forEach(auctionTransaction => commit('addAuctionToStorage', auctionTransaction));
+            collectedAuctions.forEach(auction => commit('addAuctionToStorage', auction));
```

Comment:
Why do you need to work with them separately here? The main rule is: frontend doesn't need to know anything about auctions internal logic, it should just call fetch, enrich, big, etc. So I would rather move this part into enrich function?